financeextras.php

```diff
@@ -78,3 +78,34 @@ function financeextras_civicrm_upgrade($op, CRM_Queue_Queue $queue = NULL) {
 function financeextras_civicrm_entityTypes(&$entityTypes) {
   _financeextras_civix_civicrm_entityTypes($entityTypes);
 }
+
+/**
+ * Implements hook_civicrm_pageRun().
+ *
+ * @link https://docs.civicrm.org/dev/en/master/hooks/hook_civicrm_pageRun/
+ */
+function financeextras_civicrm_pageRun($page) {
+  $hooks = [
```

Comment:
I know this approach is used in some extensions, but I really don't like it, it initialize a lot of unnecessary objects on every CiviCRM page, I would do this check:

```
$page instanceof CRM_Contribute_Page_Tab && $page->_action == CRM_Core_Action::BROWSE
```

before I initialize the object and call its `handle` method.